### DISCUSSION TITLE
Searching for all caps using regex is broken

### DISCUSSION THREAD

**Discussion by kendallm** [community/community#106798] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

When using code search to look for all caps words in our repo using regex it just returns all words, without respecting the casing in the regex.

Reproduction steps
- Search in a repo looking for only words in all caps
    - https://github.com/search?q=repo%3Akendallm%2Firis+%2F%5BA-Z%5D%2B%2F&type=code
    - `repo:kendallm/iris /[A-Z]+/`
- See search results (matching all words, not only all caps words)

Its really odd that search allows for regex searches, but does not actually use regex as defined in the search param.

**Comment by jsoref** [comment]
I'm pretty sure this is a design decision based on the implementation. I've filed:
- [ ] https://github.com/github/docs/issues/31530

To ask that it be documented.

**Comment by mike-clark-8192** [comment]
You can use `/(?-i)[A-Z]+/` to get a case-sensitive search for `[A-Z]+`.

`(?-i)` is an inline mode-specifier (`(?)`) that turns off (`-`) case insensitivity (`i`).

It is not unexpected for `[A-Z]+` in a case-insensitive regex to match lowercase characters.
cf. JavaScript with `i`: `/[A-Z]+/i`.
